### PR TITLE
Adding missing SnapshotOptions method

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
I missed these in the merge between the nullable and the SnapshotOptions PR.

### CODE REVIEW

File: Firestore/Source/Public/FIRDocumentSnapshot.h [hunk 1]
```diff
@@ -180,10 +180,22 @@ NS_SWIFT_NAME(QueryDocumentSnapshot)
...
 - (NSDictionary<NSString *, id> *)data;
 
+/**
+ * Retrieves all fields in the document as a `Dictionary`.
+ *
+ * @param options `SnapshotOptions` to configure how data is returned from the snapshot (e.g. the
+ *      desired behavior for server timestamps that have not yet been set to their final value).
+ * @return A `Dictionary` containing all fields in the document.
+ */
+- (NSDictionary<NSString *, id> *)dataWithOptions:(FIRSnapshotOptions *)options;
```
Comment (wilhuff) [inline Firestore/Source/Public/FIRDocumentSnapshot.h]: Please add a swift-language usage of this API to verify that it looks like what we want after conversion to swift.

Comment (schmidt-sebastian) [reply Firestore/Source/Public/FIRDocumentSnapshot.h]: Done. Also added an NSSwiftName for FIRServerTimestampBehavior to make the type cleaner when it shows up in code completion.

Comment (wilhuff) [reply Firestore/Source/Public/FIRDocumentSnapshot.h]: data(options:) likely. @ryanwilson, thoughts?

Comment (wilhuff) [reply Firestore/Source/Public/FIRDocumentSnapshot.h]: Makes sense. The fluency argument is a good one. That makes this change good as-is.

**Pull request merged**